pkg/prometheus/agent/statefulset_test.go

```diff
@@ -52,75 +52,29 @@ func TestListenTLS(t *testing.T) {
 }
 
 func TestWALCompression(t *testing.T) {
-	tests := []struct {
-		version       string
-		enabled       *bool
-		expectedArg   string
-		shouldContain bool
-	}{
-		// Nil should not have either flag.
-		{"v2.30.0", ptr.To(false), "--storage.agent.wal-compression", false},
-		{"v2.32.0", nil, "--storage.agent.wal-compression", false},
-		{"v2.32.0", ptr.To(false), "--no-storage.agent.wal-compression", true},
-		{"v2.32.0", ptr.To(true), "--storage.agent.wal-compression", true},
-	}
```

Comment:
Hmmmm, I'm assuming we made similar changes in other tests because the test cases array was taking waaaay more space than the test logic itself. For this test in particular it doesn't seem like it. Could you explain your thoughts when making the changes here?